Source/WebCore/platform/mediarecorder/MediaRecorderPrivateEncoder.cpp

```diff
@@ -664,26 +679,17 @@ void MediaRecorderPrivateEncoder::enqueueCompressedVideoFrame(VideoEncoder::Enco
         return;
     }
 
-    ASSERT(m_videoTrackInfo && m_videoFormatDescription);
+    ASSERT(m_videoTrackInfo);
     m_lastReceivedCompressedVideoFrame = MediaTime(frame.timestamp, 1000000);
     ASSERT(m_lastReceivedCompressedVideoFrame <= m_lastEnqueuedRawVideoFrame);
 
-    MediaSamplesBlock block;
-    block.setInfo(m_videoTrackInfo.copyRef());
-    block.append({
+    MediaSamplesBlock::SamplesVector vector(1, {
         .presentationTime = m_lastReceivedCompressedVideoFrame,
         .decodeTime = m_lastReceivedCompressedVideoFrame,
-        .data = SharedBuffer::create(WTFMove(frame.data)),
+        .data = SharedBuffer::create(frame.data),
         .flags = frame.isKeyFrame ? MediaSample::SampleFlags::IsSync : MediaSample::SampleFlags::None
     });
-    auto sample = toCMSampleBuffer(WTFMove(block), m_videoFormatDescription.get());
-    if (!sample) {
-        LOG(MediaStream, "appendVideoFrame: error creation compressed sample");
-        return;
-    }
-    m_encodedVideoFrames.append(MediaSampleAVFObjC::create(sample->get(), *m_videoTrackIndex));
-    Ref sampleObjC = m_encodedVideoFrames.last();
-    ASSERT(frame.isKeyFrame == sampleObjC->isSync());
+    m_encodedVideoFrames.append(makeUniqueRef<MediaSamplesBlock>(m_videoTrackInfo.get(), WTFMove(vector)));
```

Comment:
It is not really clear to me why m_encodedVideoFrames should own `UniqueRef<MediaSamplesBlock>` instead of `MediaSamplesBlock`.

Reply:
We maintain 3 Deque which then gets copied into another for transfer over IPC. 

A UniqueRef actually gets moved; then just 8 bytes being moved after each transfer. 

An empty MediaSamplesBlock is 32 bytes and when you take one MSB from.tbe queue to put into another, it doesn't "move" as such; it performs a deep copy. 

Just nowhere as efficient as using a Deque of UniqueRef

Additionally ; it makes the move from RetainPtr<CMSampleBuffer> to UniqueRef<MediaSamplesBlock> much easier and a very straightforward mechanical code change.